swap/src/cli/cancel_and_refund.rs

```diff
@@ -20,16 +19,19 @@ pub async fn cancel_and_refund(
         Ok(s) => s,
         Err(e) => bail!(e),
     };
-
-    tracing::info!("Refund transaction submitted");
+    if matches!(state, BobState::BtcRefunded { .. }) {
+        //Print success message only if refund succeeded.
+        tracing::info!("Refund transaction submitted");
+    }
     Ok(state)
 }
 
 pub async fn cancel(
     swap_id: Uuid,
     bitcoin_wallet: Arc<Wallet>,
     db: Arc<dyn Database + Send + Sync>,
-) -> Result<(Txid, Subscription, BobState)> {
+) -> Result<(Txid, BobState)> {
+    // Subscription was introduced in commit f39e1de964f7e2f81f9624043834a31154b68a55, but isn't used anywhere. We can't get transaction if Alice already cancelled the swap.
```

Comment:
In case tx_cancel already exists, the function can't return a subscription from submit_tx_cancel because Alice has already broadcast tx_cancel. I grepped the code and the subscription is not used anywhere.

Reply:
Ok I understand that makes sense. But please remove the comment. It's not really useful now that this has been removed. It's better to put it in the commit message.

Reply:
> Ok I understand that makes sense. But please remove the comment. It's not really useful now that this has been removed. It's better to put it in the commit message.

Done.